### ISSUE TITLE
The kernel always crash when using augmentation in model.train

### ISSUE THREAD

**Issue by ChiaoSun** [matterport/Mask_RCNN#379]
I passing the augmentation = imgaug.augmenters.Affine(rotate=(-45, 45)) in model.train() but the kernel always crash when in 2018 bowl dataset.

But it works in train_shapes.ipynb. Did the 2018 bowl dataset too big or any suggestion?

My environment are shown below:
two 1080ti 
tensorflow-gpu==1.3.0
CUDA 8
cudnn 6

**Comment by waleedka** [comment]
Does it crash with any augmentation, or just with this particular augmentation?

Maybe you can also test if the image size is the issue. In the Config class, change IMAGE_MIN_DIM and IMAGE_MAX_DIM to something small, such as 128, 128 and see if the problem still happens.

**Comment by mpsampat** [comment]
Waleed, I have the same issue. I have some additional observations; 
crashes with any augmentation (like Fliplr) and also with IMAGE_MAX_DIM as 128; 
also does not crash if we set USE_MINI_MASK = False. 
I was debugging it and I found the crash is comping from this method:
https://github.com/matterport/Mask_RCNN/blob/master/model.py#L1260
(I just did a basic debug by putting a lot of print statements :-))
I will let you know if i can pinpoint the issue.

**Comment by mpsampat** [comment]
Waleed, I traced it further to here: 
https://github.com/matterport/Mask_RCNN/blob/master/utils.py#L471
I think is it happening when the size of variable "m" is very small.

**Comment by waleedka** [comment]
Mehul Wow! great find. This is clearly a bug in skimage. I ran the 3 lines of code you sent and it crashed on my server as well. The error was: Floating point exception. I'll try to find a work-around, and I'm open to suggestions.

**Comment by mpsampat** [comment]
Thanks Waleed. Yes i have the same error Floating point exception.
This is one work-around that resolves the issue. Basically the idea is to make is (2,2) if it is (1,2) or (2,1) :
        if m.shape[0] == 1:
            m = np.matlib.repmat(m,2,1)
        if m.shape[1] == 1:
            m = np.matlib.repmat(m,1,2)
        m = skimage.transform.resize(m, mini_shape, order=1, mode="reflect",preserve_range=True)

But I am not sure why "m" is (1,2) in the first place ? is it correct that "m" is created by imgaug ? 
Maybe we should look at some masks created by imgaug to see why we have this issue. it occurs less frequently with Fliplr and more frequently with Fliplr and Affine.

**Comment by waleedka** [comment]
Thanks, @mpsampat. I found that removing `mode="reflect"` fixes the problem. And, actually, `reflect` is not the best option here anyway. The default mode, `constant`, is a better option. So fixing the mode fixes both issues. 

I just pushed a fix for this. And I [reported the bug](https://github.com/scikit-image/scikit-image/issues/3001) to scikit-image.
